etcdserver/server.go

```diff
@@ -1024,8 +1024,37 @@ func (s *EtcdServer) parseProposeCtxErr(err error, start time.Time) error {
 		if start.After(prevLeadLost) && start.Before(curLeadElected) {
 			return ErrTimeoutDueToLeaderFail
 		}
+
+		lead := types.ID(atomic.LoadUint64(&s.r.lead))
+		switch lead {
+		case types.ID(raft.None):
+			// TODO: return error to specify it happens because the cluster cannot elect leader
+		case s.ID():
+			if s.isQuorumLosed(start) {
```

Comment:
Yes, in some probability.

The proposal may get committed or dropped. There are two classical examples: 1. all followers are killed and restarted later, then it will still be leader and the proposal will be committed 2. leader is isolated, so all followers elect new leader. when the leader connects back, the proposal will be removed due to conflict log

My concern is that the check may be loose. If follower doesn't need to catch up the log/receive snapshot, the leader only needs one heartbeat to append the proposal. I set it loose to cover more possible cases.